client/question/question.html

```diff
@@ -69,6 +69,7 @@
     <script src="services/ServerHandlerService.js"></script>
     <script src="services/SessionIdService.js"></script>
     <script src="services/SolutionHandlerService.js"></script>
+    <script src="services/TipsGeneratorService.js"></script>
```

Comment:
Should this be TipGenerator? Any reason for the plural?

Reply:
This is deliberate. We generally provide one piece of feedback and one reinforcement object per submission. However, a submission may elicit multiple tips, and the functions in the service definition return a list of tips, not just one.

Reply:
Fair enough.